specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2021-06-01-preview/databoxedge.json

```diff
@@ -0,0 +1,9259 @@
+          "type": "string"
+        },
+        "resourceGroupName": {
+          "description": "Arc resource group name",
+          "type": "string"
+        },
+        "resourceName": {
+          "description": "Arc resource Name",
+          "type": "string"
+        },
+        "resourceLocation": {
+          "description": "Arc resource location",
+          "type": "string"
+        },
+        "version": {
+          "description": "Arc resource version",
+          "type": "string",
+          "readOnly": true
+        },
+        "hostPlatform": {
+          "description": "Host OS supported by the Arc addon.",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "PlatformType",
+            "modelAsString": true
+          }
+        },
+        "hostPlatformType": {
+          "description": "Platform where the runtime is hosted.",
+          "enum": [
+            "KubernetesCluster",
+            "LinuxVM"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "HostPlatformType",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "description": "Addon Provisioning State",
+          "enum": [
+            "Invalid",
+            "Creating",
+            "Created",
+            "Updating",
+            "Reconfiguring",
+            "Failed",
+            "Deleting"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "AddonState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ARMBaseModel": {
+      "description": "Represents the base class for all object models.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The path ID that uniquely identifies the object.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The object name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The hierarchical type of the object.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "AsymmetricEncryptedSecret": {
+      "description": "Represent the secrets intended for encryption with asymmetric key pair.",
+      "required": [
+        "value",
+        "encryptionAlgorithm"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The value of the secret.",
+          "type": "string"
+        },
+        "encryptionCertThumbprint": {
+          "description": "Thumbprint certificate used to encrypt \\\"Value\\\". If the value is unencrypted, it will be null.",
+          "x-ms-secret": true,
```

Comment:
Is there a reason why a certificate thumbprint is being considered as a secret?

Reply:
Thanks @paulomarquesc.  This is an existing contracts for all secrets in ASE. Thumbprint is mostly not used internally in the device. I guess this is format we carried forwarded from our existing contracts.

Reply:
@paulomarquesc , can you please let me know if the changes looks fine and  Can the PR be merged? Thanks

Reply:
Thanks Sameer, got it, this RP is the only one considering a cert thumbprint and I believe this should be fixed. Looping @j5lim since he is the primary ARM oncall this week, do you agree with this ?

Reply:
Thumbprint should not be marked as a secret. Azure platform features rely on x-ms-secret not to reveal it in a plan text. If you mark this as a secret, thumbprint may not be visible where it should be.